### PR TITLE
add cli and library getting started guides

### PR DESCRIPTION
**Pull request by tomerd**
motivation: getting started guides

changes:
* add a cli guide
* add a library guide + link from index page
* share install instructions across pages

### CODE REVIEW

**Comment by tomerd** [comment]
this is ready for initial review

File: getting-started/cli-tool/index.md [hunk 1]
```diff
@@ -3,4 +3,189 @@ layout: page
...
+| | | |   ___  | | | |   ___          ___  __      __ (_)  / _| | |_  | |
+| |_| |  / _ \ | | | |  / _ \        / __| \ \ /\ / / | | | |_  | __| | |
+|  _  | |  __/ | | | | | (_) |  _    \__ \  \ V  V /  | | |  _| | |_  |_|
+|_| |_|  \___| |_| |_|  \___/  ( )   |___/   \_/\_/   |_| |_|    \__| (_)
+                              |/
+~~~
+
+## Argument parsing
+
+Most command line tools need to be able and parse command line arguments.
```
Comment (kaishin) [inline getting-started/cli-tool/index.md:125]:
```suggestion
Most command line tools need to be able to parse command line arguments.
```

File: getting-started/cli-tool/index.md [hunk 2]
```diff
@@ -3,4 +3,189 @@ layout: page
...
+    ),
+    .testTarget(
+      name: "hello-swiftTests",
+      dependencies: ["hello-swift"]
+    ),
+  ]
+)
+~~~
+
+We can now import the argument parsing module provided by `swift-argument-parser` and use it in our application
```
Comment (kaishin) [inline getting-started/cli-tool/index.md:158]:
```suggestion
We can now import the argument parsing module provided by `swift-argument-parser` and use it in our application:
```

File: getting-started/library/index.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    self.description = emailString
+  }
+}
+
+private struct InvalidEmailError: Error {
+  let email: String
+}
+~~~
+
+Now lets add a unit test for this strongly types Email API.
```
Comment (kaishin) [inline getting-started/library/index.md:83]:
```suggestion
Now lets add a unit test for this strongly typed Email API.  
```

**Review by kaishin — APPROVED** [review]
Really helpful introductory guides :) Some minor typos to fix aside, looks good to go.

**Comment by tomerd** [comment]
@0xTim your turn!

**Pull request merged**